Draft: java okta email idx guide (#3306)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/authenticators-okta-email/main/java/integratechallengemagiclink.md

### 1. Authenticate the user credentials

> After a user has initiated the sign-in process by entering the username and password, create an `AuthenticationOptions` object and set its `username` and `password` fields to the values set by the user. Pass this object as a parameter to the `authenticate` method on the `IdxAuthenticationWrapper` that you have instantiated.

Reviewer: ```suggestion
After a user starts the sign-in process by entering the username and password, create an `AuthenticationOptions` object and set its `username` and `password` fields to the values set by the user. Pass this object as a parameter to the `authenticate()` method on the `IdxAuthenticationWrapper` that you have instantiated.
```

```java
// Begin the authentication flow
AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.begin();

> // TODO: add device info

Reviewer: Delete the TODO line

* `AWAITING_AUTHENTICATOR_VERIFICATION` that is covered in a later section

### 3. Display a list of possible authenticators

You can find the names of the available authenticators for enrollment or challenge in the `AuthenticationResponse` object's `authenticators` collection. Display all of the authenticators that the user has enrolled and are ready for use.

```java
case AWAITING_AUTHENTICATOR_SELECTION:
    List<Authenticator> authenticators = authenticationResponse.getAuthenticators();

    // show authenticators to user
    IntStream.range(0, authenticators.size()).forEach(index -> {
        console.writer().println(index + " - " + authenticators.get(index).getLabel());
    });

    // prompt user to select a factor
    String selection = console.readLine("Select an Authenticator: ");
    Authenticator authenticator = authenticators.get(Integer.parseInt(selection));

    // tell Okta which Authenticator is used
    return idxAuthenticationWrapper.selectAuthenticator(proceedContext,  authenticator);
```

When the email authenticator is selected, an email will be sent to the user.

### 4. Check authenticator status

> The next authentication status will be `AWAITING_AUTHENTICATOR_VERIFICATION` which indicates information is needed. A client can either accept a TOTP code from the email address or poll until the user has completed the flow in different browser window.

Reviewer: ```suggestion
The next authentication status will be `AWAITING_AUTHENTICATOR_VERIFICATION` which indicates information is needed. A client can either accept a TOTP code from the email address or poll until the user has completed the flow in a different browser window.
```

### 5. Display OTP input page or poll

> The user needs to leave your application and check their email. While they are doing this your application can continue to poll Okta, and accept the input of a TOTP code:

Reviewer 2: ```suggestion
The user needs to leave your application and check their email. While they are doing this, your application can continue to poll Okta, and accept the input of a TOTP code:
```
